api/src/components/AddProduct.tsx

```diff
@@ -0,0 +1,98 @@
+
+import * as React from 'react';
+import Button from '@mui/material/Button';
+import TextField from '@mui/material/TextField';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogContentText from '@mui/material/DialogContentText';
+import DialogTitle from '@mui/material/DialogTitle';
+import {useEffect, useState} from 'react';
+import axios from 'axios';
+
+export default function FormDialog() {
+  const [open, setOpen] = useState(false);
+  const [title,setTitle] = useState('');
+  const [description, setDescription] = useState('');
+
+const handleDescriptionChange = (event) => {
+  setDescription(event.target.value)
+}
+console.log(description)
+
+  const handleTitleChange = (event) => {
+    setTitle(event.target.value)
+  }
+  console.log(title);
+
+  const handleClickOpen = () => {
+    setOpen(true);
+  };
+
+  const handleClose = () => {
```

Comment:
It doesn't make any sense to perform a sideEffect inside the handleClose()
You should only close the dialog inside handleClose(), no useEffects for data manipulations should be present